core/src/test/java/hudson/util/RobustReflectionConverterTest.java

```diff
@@ -108,13 +124,16 @@ public void classOwnership() {
         });
         String prefix1 = RobustReflectionConverterTest.class.getName() + "_-";
         String prefix2 = RobustReflectionConverterTest.class.getName() + "$";
-        assertEquals("<Projekt><bildz><Bild><steppes>"
+        String target = "<Projekt><bildz><Bild><steppes>"
                 + "<Enchufla plugin='p1'><number>1</number><direction>North</direction></Enchufla>"
                 // note no plugin='p2' on <boot/> since that would be redundant; <jacket/> is quiet even though unowned
                 + "<Moonwalk plugin='p2'><number>2</number><boot/><lover class='Billy' plugin='p3'/></Moonwalk>"
                 + "<Moonwalk plugin='p2'><number>3</number><boot/><jacket/><lover class='Jean' plugin='p4'/></Moonwalk>"
-                + "</steppes></Bild></bildz></Projekt>",
-                xs.toXML(p).replace(prefix1, "").replace(prefix2, "").replaceAll("\r?\n *", "").replace('"', '\''));
+                + "</steppes></Bild></bildz></Projekt>";
+        String sample = xs.toXML(p).replace(prefix1, "").replace(prefix2, "").replaceAll("\r?\n *", "").replace('"', '\'');
+        Map<String, String> targetClassOwnerMap = getClassOwnerMap(target);
+        Map<String, String> sampleClassOwnerMap = getClassOwnerMap(sample);
+        assertTrue(targetClassOwnerMap.equals(sampleClassOwnerMap));
```

Comment:
This would fail with an unhelpful message if the assertion is violated. Better to use `assertEquals` or Hamcrest matchers for a better error message.